test/e2e_node/pod_resize_test.go

```diff
@@ -307,56 +309,161 @@ func verifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
+					}
+					// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+					// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+					expectedCPUShares = int64(1 + ((expectedCPUShares-2)*9999)/262142)
+				}
+				if expectedMemLimitString != "0" {
+					err := verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString)
+					if err != nil {
+						return err
+					}
+				}
+				err := verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString)
+				if err != nil {
+					return err
+				}
+				err = verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10))
 				if err != nil {
 					return err
 				}
 			}
-			err := verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString)
-			if err != nil {
-				return err
+		}
+
+		// Accumulate container resources for verifying pod
+		if cpuRequest != nil {
+			podCPURequestMilliValue += cpuRequest.MilliValue()
+		}
+		if podCPULimitMilliValue >= 0 {
+			if cpuLimit == nil || cpuLimit.IsZero() {
+				podCPULimitMilliValue = -1
+			} else {
+				podCPULimitMilliValue += cpuLimit.MilliValue()
+			}
+		}
+		if podMemoryLimitInBytes >= 0 {
+			if expectedMemLimitInBytes == 0 {
+				podMemoryLimitInBytes = -1
+			} else {
+				podMemoryLimitInBytes += expectedMemLimitInBytes
 			}
-			err = verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10))
-			if err != nil {
+		}
+	}
+
+	// Verify pod cgroup
+
+	verifyPodCgroupValue := func(cgPath, expectedCgValue string) error {
+		cgValueBytes, err := os.ReadFile(cgPath)
+		if err != nil {
+			return fmt.Errorf("failed to read pod cgroup value %q", cgPath)
+		}
+		cgValue := strings.Trim(string(cgValueBytes), "\n")
+		if cgValue != expectedCgValue {
+			return fmt.Errorf("pod cgroup value '%s' not equal to expected '%s'", cgValue, expectedCgValue)
+		}
+		return nil
+
+	}
+
+	expectedPodCPUShares := int64(kubecm.MilliCPUToShares(podCPURequestMilliValue))
+	expectedPodCPUQuota := int64(-1)
+	if podCPULimitMilliValue >= 0 {
+		expectedPodCPUQuota = kubecm.MilliCPUToQuota(podCPULimitMilliValue, kubecm.QuotaPeriod)
+	}
+	expectedPodCPULimitString := strconv.FormatInt(expectedPodCPUQuota, 10)
+	if podMemoryLimitInBytes < 0 {
+		podMemoryLimitInBytes = 0
+	}
+	expectedPodMemLimitString := strconv.FormatInt(podMemoryLimitInBytes, 10)
+	if podOnCgroupv2Node {
+		if expectedPodCPULimitString == "-1" {
+			expectedPodCPULimitString = "max"
+		}
+		expectedPodCPULimitString = fmt.Sprintf("%s %s", expectedPodCPULimitString, CPUPeriod)
+		if expectedPodMemLimitString == "0" {
+			expectedPodMemLimitString = "max"
+		}
+		// convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight value
+		// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+		expectedPodCPUShares = int64(1 + ((expectedPodCPUShares-2)*9999)/262142)
```

Comment:
nit: redundant type conversion
```suggestion
		expectedPodCPUShares = 1 + ((expectedPodCPUShares-2)*9999)/262142
```